File: PersianCalendar/src/main/java/com/github/praytimes/MinuteOrAngleDouble.java
package com.github.praytimes;

public class MinuteOrAngleDouble {
    private final boolean isMin;
    private final double value;

    public MinuteOrAngleDouble(double value, boolean isMin) {
        this.value = value;
        this.isMin = isMin;
    }

    public boolean isMin() {
        return isMin;
    }

    public double getValue() {
        return value;
    }
}

File: shared/src/commonMain/kotlin/io/github/persiancalendar/praytimes/MinuteOrAngleDouble.kt
package io.github.persiancalendar.praytimes

internal data class MinuteOrAngleDouble(val value: Double, val isMinutes: Boolean)

internal val Number.deg get() = MinuteOrAngleDouble(this.toDouble(), false)
internal val Int.min get() = MinuteOrAngleDouble(this.toDouble(), true)
